core/new-gui/src/app/workspace/component/property-editor/operator-property-edit-frame/operator-property-edit-frame.component.ts

```diff
@@ -483,6 +483,79 @@ export class OperatorPropertyEditFrameComponent implements OnInit, OnChanges, On
         };
       }
 
+      if (isDefined(mapSource.attributeType)) {
+        // TODO: Add instead of replace
```

Comment:
To improve readability, try to break the large logic into small helper functions, and avoid using multiple exit points (return) for one function. Also, please leave some comments.

Reply:
Created two helper functions. But I'm not sure how to avoid using the type guard conditions, as some variables may be "undefined", and I need to check that before doing something.